### PR TITLE
Add kube_replicationcontroller_owner

### LINKED ISSUE #1055: Add metric for owner of ReplicationController - kube_replicationcontroller_owner
We do not have a metric with the information about the owner of `ReplicationController`. 

/kind feature

### PR DESCRIPTION
**Pull request by olivierlemasle**
**What this PR does / why we need it**:

Add metric for owner of ReplicationController - `kube_replicationcontroller_owner`

**Which issue(s) this PR fixes**:
Fixes #1055

### CODE REVIEW

File: internal/store/replicationcontroller.go [hunk 1]
```diff
@@ -152,6 +154,44 @@ var (
...
+					})
+				} else {
+					for _, owner := range owners {
+						if owner.Controller != nil {
+							ms = append(ms, &metric.Metric{
+								LabelKeys:   labelKeys,
+								LabelValues: []string{owner.Kind, owner.Name, strconv.FormatBool(*owner.Controller)},
+								Value:       1,
+							})
+						} else {
```
Comment (lilic) [inline internal/store/replicationcontroller.go]: Do you mind adding a test case for when it is nil, thanks!

Comment (olivierlemasle) [reply internal/store/replicationcontroller.go]: Sure!

File: docs/replicationcontroller-metrics.md [hunk 2]
```diff
@@ -10,3 +10,4 @@
 | kube_replicationcontroller_spec_replicas | Gauge | `replicationcontroller`=&lt;replicationcontroller-name&gt; <br> `namespace`=&lt;replicationcontroller-namespace&gt; | STABLE |
 | kube_replicationcontroller_metadata_generation | Gauge | `replicationcontroller`=&lt;replicationcontroller-name&gt; <br> `namespace`=&lt;replicationcontroller-namespace&gt; | STABLE |
 | kube_replicationcontroller_created | Gauge | `replicationcontroller`=&lt;replicationcontroller-name&gt; <br> `namespace`=&lt;replicationcontroller-namespace&gt; | STABLE |
+| kube_replicationcontroller_owner | Gauge | `replicationcontroller`=&lt;replicationcontroller-name&gt; <br> `namespace`=&lt;replicationcontroller-namespace&gt; <br> `owner_kind`=&lt;owner kind&gt; <br> `owner_name`=&lt;owner name&gt; <br> `owner_is_controller`=&lt;whether owner is controller&gt;  | STABLE |
```
[the remark this answers is not in the document]
Comment (olivierlemasle) [reply docs/replicationcontroller-metrics.md]: Ok. I actually followed what had been done in the PR for [kube_job_owner](https://github.com/kubernetes/kube-state-metrics/pull/681/files) and it landed as STABLE.

@lilic  What is the policy regarding this metric's stage (e.g. when does a metric land to STABLE)? In [README documentation](https://github.com/kubernetes/kube-state-metrics/blob/f8bcc024791126e1a8b3f7d07e4392a5373b66e7/docs/README.md#metrics-stages), it looks as if EXPERIMENTAL means the underlying K8s object is experimental, not the metric, but I may misunderstand.

[the remark this answers is not in the document]
Comment (lilic) [reply docs/replicationcontroller-metrics.md]: yes, we had a discussion about this and have an issue open, we said we want to start by adding them as EXPERIMENTAL for 2.0. It doesn't hurt :)

File: internal/store/replicationcontroller.go [hunk 3]
```diff
@@ -152,6 +154,44 @@ var (
...
+							ms = append(ms, &metric.Metric{
+								LabelKeys:   labelKeys,
+								LabelValues: []string{owner.Kind, owner.Name, "false"},
+								Value:       1,
+							})
+						}
+					}
+				}
+
```
Comment (tariq1890) [inline internal/store/replicationcontroller.go]: I think we can address the code duplication here. Let's have the if loop to just set the string variable with the booleanStrFormal value and then append the metric to the metricFamily slice.

Comment (olivierlemasle) [reply internal/store/replicationcontroller.go]: Sure!

File: docs/replicationcontroller-metrics.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (olivierlemasle) [reply docs/replicationcontroller-metrics.md]: Thanks, I found the open issue with this discussion.

**Comment by olivierlemasle** [comment]
I've just rebased and addressed @tariq1890's comment.

**Comment by tariq1890** [comment]
Thanks @olivierlemasle :)!
